geode-core/src/main/java/org/apache/geode/cache/client/internal/OpExecutorImpl.java

```diff
@@ -251,21 +251,8 @@ private Object executeWithServerAffinity(ServerLocation loc, Op op) {
     int transactionId = absOp.getMessage().getTransactionId();
     // for CommitOp we do not have transactionId in AbstractOp
     // so set it explicitly for TXFailoverOp
-    try {
```

Comment:
Is the goal here to get rid of the retry attempt entirely? Does it inherently lead to unexpected behavior and should have never been done in the first case?

Reply:
It does retry once; and continues the operation if the TxId is found; if id is not found it throws the exception back so that application can handle it...

Reply:
The TXFailover is used when a client lost connection to a server. The client will fail over to another server, and the TXFailover is for the new server to find if the original server who hosting the transaction still exist. If it exist, it will let client continue the transaction. 

This fix only eliminates the scenario when the new server does not find any server hosting the transaction but client finds it is the first operation of the transaction and does the retry instead of failing the transaction.

Reply:
New changes incorporates the review comments for test.